api/envoy/config/core/v3/grpc_service.proto

```diff
@@ -55,6 +56,14 @@ message GrpcService {
     // This limit is applied to individual messages in the streaming response and not the total size of streaming response.
     // Defaults to 0, which means unlimited.
     google.protobuf.UInt32Value max_receive_message_length = 4;
+
+    // This provides gRPC client level control over envoy generated headers.
+    // For ``x-envoy-internal`` and ``x-forwarded-for`` header:
+    // If true, the header will be sent. But it can be overridden by setting the ``send_internal`` or ``send_xff``
+    // to false in ``Http::AsyncClient::StreamOptions``, as per stream-level control.
+    // If false, the header will be removed and can not be overridden by per stream option.
+    // Default to true.
+    google.protobuf.BoolValue send_envoy_generated_header = 5;
```

Comment:
I think invert this and use `skip_envoy_headers` as the name for consistency with HCM and XFF (for example).

Reply:
Sure done.